Update docs about list indentation

lydell opened the pull request.

lydell committed:
Update docs about list indentation

waylan commented on docs/index.md:
```diff
@@ -91,7 +92,9 @@ are summarized below:
     desired. Be warned however, as this will affect indentation for all aspects
-    of the syntax (including root level code blocks).
+    of the syntax (including root level code blocks). Alternatively, you could
+    use the [mdx_truly_sane_lists](https://github.com/radude/mdx_truly_sane_lists)
```
While I understand the spirit of this change, I'd rather not favor any one third-party extension over another. Instead, this link should point to the [wiki page][1] where we keep a list of all known third party extensions. Perhaps one of them will meet the users needs.

And that list is publicly editable, so if you see an extension is missing, feel free to add it yourself.

[1]: https://github.com/waylan/Python-Markdown/wiki/Third-Party-Extensions

waylan commented on docs/index.md:
```diff
@@ -80,7 +80,8 @@ are summarized below:
     subsequent paragraph in a list item **must** be indented by either 4 spaces
-    or one tab" (emphasis added). However, many implementations do not enforce
+    or one tab" (emphasis added). However, many implementations – including the
+    original markdown.pl implementation itself – do not enforce
```
I see no value in adding this. Lets leave it out.

lydell replied:
#125, #313, #442 and #751 mention the original implementation as a reason for Python-Markdown to also implement this, so I figured it'd be helpful to mention that we already know that markdown.pl already behaves the way it does to avoid duplicate issues.

waylan replied:
People should be searching old issues before opening a new issue, in which case they would know that. I realize GitHub doesn't search closed issues by default, but we call this out in our (new) [contributing guide][1]. If people follow the guide, then this is a non-issue.

[1]: https://github.com/Python-Markdown/markdown/blob/master/docs/contributing.md#issues

lydell replied:
People don't. I know, because I didn't (until afterwards).

waylan reviewed (changes requested):
Thanks for your contribution. I have noted some suggestions below. If you make the requested changes, I'll merge your PR.

lydell committed:
Link to extensions wiki page instead

waylan committed:
Follow Contributing Guide

As per the guide, we should only fix the area we are working on, so we leave the rest of the document as is, but follow the guide with our change and use a reference style link with the reference at the end of the document.

Also, remove unhelpful comment.

lydell commented:
It was worth a try! Thanks for your time! :tada:
